common/gce.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for using the Google Compute Engine (GCE) API."""
+import threading
+
+import dateutil.parser
+
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+
+thread_local = threading.local()  # pylint: disable=invalid-name
+
+
+def initialize():
+    """Initialize the thread-local configuration with things we need to use the
+    GCE API."""
+    thread_local.credentials = GoogleCredentials.get_application_default()
```

Comment:
This does not need to be in thread local ? just credentials = ?

Reply:
A lot of Google libraries break when used in a multithreaded environment. Since we use threads in FuzzBench, I'm making it thread local. This is similar to how we use locks in database/utils.py but here we don't need to worry about connection limits (unlike in db_utils).

Reply:
I meant, you only need to do thread_local.service, not thread_local.credentials